src/esp/assets/GenericSemanticMeshData.cpp

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "GenericSemanticMeshData.h"
+
+#include <Corrade/Containers/Array.h>
+#include <Corrade/Containers/ArrayView.h>
+#include <Corrade/Containers/ArrayViewStl.h>
+#include <Corrade/Utility/Algorithms.h>
+#include <Corrade/Utility/FormatStl.h>
+#include <Magnum/DebugTools/ColorMap.h>
+#include <Magnum/Math/FunctionsBatch.h>
+#include <Magnum/Math/PackingBatch.h>
+#include <Magnum/MeshTools/Interleave.h>
+#include <Magnum/MeshTools/RemoveDuplicates.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Shaders/GenericGL.h>
+
+#include "esp/scene/HM3DSemanticScene.h"
+
+namespace Cr = Corrade;
+namespace Mn = Magnum;
+
+namespace esp {
+namespace assets {
+
+std::vector<std::unique_ptr<GenericSemanticMeshData>>
+GenericSemanticMeshData::buildSemanticMeshData(
+    const Cr::Containers::Optional<Mn::Trade::MeshData>& srcMeshData,
```

Comment:
No `Optional` here please :) The code of the function assumes the optional is non-null anyway.